embassy-rp/src/pwm.rs

```diff
@@ -81,24 +81,23 @@ impl From<InputMode> for Divmode {
 }
 
 /// PWM driver.
-pub struct Pwm<'d, T: Slice> {
-    inner: PeripheralRef<'d, T>,
+pub struct Pwm<'d> {
+    inner: PeripheralRef<'d, pac::pwm::Channel>,
```

Comment:
I think you shouldn't need htis field anymore, because you can obtain it any time from the PAC if you know the channel number in `self.channel`